README.md

```diff
@@ -2,8 +2,120 @@
 
 Expose Lucene features to erlang RPC.
 
+## Setting up the development environment
+
+We use a combination of [`direnv`](https://github.com/direnv/direnv) and [`asdf`](https://github.com/asdf-vm/asdf) to manage development environment.
+The minimal set of tools you need to install manually is:
+
+- xcode
+- brew
+- asdf
+- git
+- coreutils
+
+1. install asdf itself
+```
+brew install asdf
+```
+2. integrate asdf with your shell
+  - fish
+   ```
+   echo -e "\nsource $(brew --prefix asdf)/libexec/asdf.fish" >>  ~/.config/fish/config.fish
+   ```
+  - bash
+   ```
+   echo -e "\n. $(brew --prefix asdf)/libexec/asdf.sh" >> ~/.bash_profile
+   ```
+  - zsh
+   ```
+   echo -e "\n. $(brew --prefix asdf)/libexec/asdf.sh" >> ${ZDOTDIR:-~}/.zshrc
+   ```
+3. install direnv plugin
+```
+asdf plugin-add direnv
+asdf install direnv latest
+asdf global direnv latest
+```
+4. integrate direnv plugin with your shell
+  - fish
+   ```
+   printf "\
+   # Hook direnv into your shell.
+   asdf exec direnv hook fish | source
+
+   # A shortcut for asdf managed direnv.
+   function direnv
+     asdf exec direnv \"\$argv\"
+   end
+   " >> ~/.config/fish/config.fish
+   ```
+  - bash
+   ```
+   cat << EOF >> ~/.bashrc
+   # Hook direnv into your shell.
+   eval "$(asdf exec direnv hook bash)"
+
+   # A shortcut for asdf managed direnv.
+   direnv() { asdf exec direnv "$@"; }
+   EOF
+   ```
+  - zsh
+   ```
+   cat << EOF >>  ~/.zshrc
+   # Hook direnv into your shell.
+   eval "$(asdf exec direnv hook zsh)"
+
+   # A shortcut for asdf managed direnv.
+   direnv() { asdf exec direnv "$@"; }
+   EOF
+   ```
+5. enable `use asdf` feature
+```
+echo 'source "$(asdf direnv hook asdf)"' >> ~/.config/direnv/direnvrc
+```
+6. exit from the current shell or start a new one
+7. `cd` into the directory where you checked out `clouseau` and do `direnv allow`
+8. do `make tools` (we need it only when we add new tools into `.tool-versions`)
+
+Now every time you `cd` into the project directory the correct versions of tools
+will be activated.
+
+### Custom Java
+
+In some cases version of Java is not available in asdf-java.
+There is a trick which makes globally available version recognized by
+asdf.
+
+```
+ln -s /Library/Java/JavaVirtualMachines/jdk1.7.0_80.jdk/Contents/Home ~/.asdf/installs/java/oracle-1.7.0
+asdf reshim java
+```
+
+## Building distribution
+
+Just run `make` to get `target/clouseau-{version}-SNAPSHOT.zip` file.
+
+## Running locally
+
+To run single node instance use `make clouseau1`.
```

Comment:
Would not it better to create a `run` target and leave the name as a parameter?  Something like this:

```console
make run NAME=clouseau1
```

It could help to remove many of the hard-wired parts.

Reply:
Collapsed three targets into one in 5e95fd5c9b72d0eac92bae9a519bd115ee22ce7f

Reply:
I can add generic `make run` in addition to specific 1..3 if it is useful.